src/rust/src/hardsubx/classifier.rs

```diff
@@ -0,0 +1,202 @@
+    pub prev: *mut cc_subtitle,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct lib_hardsubx_ctx {
+    pub cc_to_stdout: ::std::os::raw::c_int,
+    pub subs_delay: i64,
+    pub last_displayed_subs_ms: i64,
+    pub basefilename: *mut ::std::os::raw::c_char,
+    pub extension: *const ::std::os::raw::c_char,
+    pub current_file: ::std::os::raw::c_int,
+    pub inputfile: *mut *mut ::std::os::raw::c_char,
+    pub num_input_files: ::std::os::raw::c_int,
+    pub system_start_time: i64,
+    pub write_format: ccx_output_format,
+    pub format_ctx: *mut AVFormatContext,
+    pub codec_ctx: *mut AVCodecContext,
+    pub codec: *mut AVCodec,
+    pub frame: *mut AVFrame,
+    pub rgb_frame: *mut AVFrame,
+    pub packet: AVPacket,
+    pub options_dict: *mut AVDictionary,
+    pub sws_ctx: *mut SwsContext,
+    pub rgb_buffer: *mut u8,
+    pub video_stream_id: ::std::os::raw::c_int,
+    pub im: *mut PIX,
+    pub tess_handle: *mut TessBaseAPI,
+    pub cur_conf: f32,
+    pub prev_conf: f32,
+    pub tickertext: ::std::os::raw::c_int,
+    pub hardsubx_and_common: ::std::os::raw::c_int,
+    pub dec_sub: *mut cc_subtitle,
+    pub ocr_mode: ::std::os::raw::c_int,
+    pub subcolor: ::std::os::raw::c_int,
+    pub min_sub_duration: f32,
+    pub detect_italics: ::std::os::raw::c_int,
+    pub conf_thresh: f32,
+    pub hue: f32,
+    pub lum_thresh: f32,
+}
+
+pub unsafe extern "C" fn get_ocr_text_simple(
+    ctx: *mut lib_hardsubx_ctx,
+    image: *mut PIX,
+) -> *mut ::std::os::raw::c_char {
+    let text_out: *mut ::std::os::raw::c_char;
+
+    TessBaseAPISetImage2((*ctx).tess_handle, image);
+
+    if TessBaseAPIRecognize((*ctx).tess_handle, null::<ETEXT_DESC>() as *mut ETEXT_DESC) != 0 {
+        mprint_mem_safe("Error in Tesseract recognition, skipping frame\n");
+        null::<c_char>() as *mut c_char
+    } else {
+        text_out = TessBaseAPIGetUTF8Text((*ctx).tess_handle);
+
+        if text_out == null::<c_char>() as *mut c_char {
+            mprint_mem_safe("Error getting text, skipping frame\n");
+        }
+        text_out
+    }
+}
+
+unsafe fn _tess_string_helper(it: *mut TessResultIterator, level: TessPageIteratorLevel) -> String {
+    // Function extracts string from tess iterator object
+    // frees memory associated with tesseract string
+    // takes and gives ownership of the string
+
+    let ts_word_ptr: *mut ::std::os::raw::c_char = TessResultIteratorGetUTF8Text(it, level);
+
+    if ts_word_ptr == null::<c_char>() as *mut c_char {
+        // this is required because trying to generate
+        // CStr from null pointer will be a segmentation fault
+        return String::new();
+    }
+
+    let ts_word = ffi::CStr::from_ptr(ts_word_ptr);
+    let ts_word_arr = ffi::CStr::to_bytes_with_nul(&ts_word);
+
+    let ts_word_string: String = match String::from_utf8(ts_word_arr.to_vec()) {
+        Ok(string_rep) => string_rep,
+        Err(error) => std::panic::panic_any(error),
+    };
```

Comment:
Can this be changed to,
```rust
unsafe {
        CStr::from_ptr(ts_word_ptr).to_string_lossy().into_owned()
    }
```

Reply:
The null check has to be there because it will fail otherwise (I have tested that eventuality) 
Otherwise yes

I will make the change.